### PR TITLE
docs: Add paragraph about @ operator

### PR DESCRIPTION
**Pull request by Secrus**
Resolves: #5136 
Resolves: #3142

Added documentation on `@` operator in dependency specification.

### CODE REVIEW

File: docs/dependency-specification.md [hunk 1]
```diff
@@ -77,10 +77,25 @@ You can specify the exact version of a package.
...
 If other dependencies require a different version, the solver will ultimately fail and abort any install or update procedures.
 
-#### Multiple requirements
+### Multiple requirements
 
 Multiple version requirements can also be separated with a comma, e.g. `>= 1.2, < 1.5`.
 
+### `@` operator
+
+You can specify requirements, using `@` operator. This is understood the same way as `==` specifier, with one difference.
```
Comment (clintonroy) [inline docs/dependency-specification.md:86]: using the '@' operator, maybe

File: docs/dependency-specification.md [hunk 2]
```diff
@@ -77,10 +77,25 @@ You can specify the exact version of a package.
...
+After `@` symbol, you can use any specifiers that are valid in `pyproject.toml` file. For example:
```
Comment (clintonroy) [inline docs/dependency-specification.md:87]: After the '@' symbol, maybe.

Also, maybe best to stick with operator or symbol throughout.

File: docs/dependency-specification.md [hunk 3]
```diff
@@ -77,10 +77,25 @@ You can specify the exact version of a package.
...
+
+```shell
+poetry add django@^4.0.0
+```
+
+Would translate to following entry in `pyproject.toml`:
```
Comment (clintonroy) [inline docs/dependency-specification.md:93]: Would translate through to the following entry in pyproject.toml

**Comment by Secrus** [comment]
@clintonroy I have addressed your comments.

**Comment by neersighted** [comment]
I'm wondering if this is really the correct place to add the docs -- since there's no surrounding mention of `poetry add` it feels out of place. If this really is the best place then the opening sentence/section title should probably be something along the lines of "adding packages with constraints using the CLI" -- if there's a more natural location with other CLI docs, we should just add a link to that section.

**Comment by Secrus** [comment]
@neersighted I added link in `add` command docs and changed first paragraph of the `@` documentation.

File: docs/dependency-specification.md [hunk 4]
```diff
@@ -84,6 +84,40 @@ You can specify the exact version of a package.
 This will tell Poetry to install this version and this version only.
 If other dependencies require a different version, the solver will ultimately fail and abort any install or update procedures.
 
+### Using the `@` operator
+
+When adding dependencies via CLI `add` command, you can use `@` operator.
```
Comment (neersighted) [inline docs/dependency-specification.md:89]:
```suggestion
When adding dependencies via CLI `add` command, you can use the `@` operator.
```

File: docs/dependency-specification.md [hunk 5]
```diff
@@ -84,6 +84,40 @@ You can specify the exact version of a package.
...
+This is understood the same way as `==` specifier, with one difference. After the `@` operator, you can use any
+specifiers that are valid in `pyproject.toml` file. For example:
```
Comment (neersighted) [inline docs/dependency-specification.md:91]:
```suggestion
This is understood similarly to the `==` syntax, but also allows prefixing any
specifiers that are valid in `pyproject.toml`. For example:
```

File: docs/cli.md [hunk 6]
```diff
@@ -304,6 +304,10 @@ poetry add "pendulum>=2.0.5"
 poetry add pendulum==2.0.5
 ```
 
+{{% note %}}
+See the [Dependency specification]({{< relref "dependency-specification#using-the--operator" >}}) page for more information about `@` operator.
```
Comment (neersighted) [inline docs/cli.md:308]:
```suggestion
See the [Dependency specification]({{< relref "dependency-specification#using-the--operator" >}}) page for more information about the `@` operator.
```

File: docs/dependency-specification.md [hunk 7]
```diff
@@ -84,6 +84,40 @@ You can specify the exact version of a package.
...
+poetry add django@^4.0.0
+```
+
+Would translate to the following entry in `pyproject.toml`:
+```toml
+Django = "^4.0.0"
+```
+
+This operator, can also be used with in conjunction with word `latest`, to specify latest available version.
+For example:
```
Comment (neersighted) [inline docs/dependency-specification.md:103]:
```suggestion
The special keyword `latest` is also understood by the `@` operator:
```

File: docs/dependency-specification.md [hunk 8]
```diff
@@ -84,6 +84,40 @@ You can specify the exact version of a package.
...
+
+When adding dependencies via CLI `add` command, you can use `@` operator.
+This is understood the same way as `==` specifier, with one difference. After the `@` operator, you can use any
+specifiers that are valid in `pyproject.toml` file. For example:
+
+```shell
+poetry add django@^4.0.0
+```
+
+Would translate to the following entry in `pyproject.toml`:
```
Comment (neersighted) [inline docs/dependency-specification.md:97]:
```suggestion
The above would translate to the following entry in `pyproject.toml`:
```

File: docs/dependency-specification.md [hunk 9]
```diff
@@ -84,6 +84,40 @@ You can specify the exact version of a package.
...
+Django = "^4.0.0"
+```
+
+This operator, can also be used with in conjunction with word `latest`, to specify latest available version.
+For example:
+```shell
+poetry add django@latest
+```
+
+Would translate to the following entry in `pyproject.toml`:
```
Comment (neersighted) [inline docs/dependency-specification.md:108]:
```suggestion
The above would translate to the following entry in `pyproject.toml`, assuming the latest release of `django` is `4.0.5`:
```

File: docs/dependency-specification.md [hunk 10]
```diff
@@ -84,6 +84,40 @@ You can specify the exact version of a package.
...
+```toml
+Django = "^4.0.5"
+```
+
+#### Extras
+
+To use extras for package using the `@` operator, formula is: `package[extra]@version`, for example:
```
Comment (neersighted) [inline docs/dependency-specification.md:115]:
```suggestion
Extras and `@` can be combined as one might expect (`package[extra]@version`):
```

**Review by neersighted — CHANGES_REQUESTED** [review]

File: docs/dependency-specification.md [hunk 11]
```diff
@@ -86,33 +86,32 @@ If other dependencies require a different version, the solver will ultimately fa
 
 ### Using the `@` operator
 
-When adding dependencies via CLI `add` command, you can use `@` operator.
-This is understood the same way as `==` specifier, with one difference. After the `@` operator, you can use any
-specifiers that are valid in `pyproject.toml` file. For example:
+When adding dependencies via CLI `add` command, you can use the `@` operator.
```
Comment (neersighted) [inline docs/dependency-specification.md:89]:
```suggestion
When adding dependencies via `poetry add`, you can use the `@` operator.
```

**Review by neersighted — APPROVED** [review]

**Pull request merged**